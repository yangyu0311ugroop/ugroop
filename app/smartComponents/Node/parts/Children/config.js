import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    ids: NODE_STORE_SELECTORS.children,
  },
  setValue: {},
};
