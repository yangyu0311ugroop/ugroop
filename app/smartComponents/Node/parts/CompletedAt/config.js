import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { RESAGA_HELPERS } from 'utils/helpers/resaga';

export const CONFIG = {
  value: {
    completedAt: RESAGA_HELPERS.subscribeIfNotGiven(
      NODE_STORE_SELECTORS.completedAt,
      'completedAt',
    ),
  },
  setValue: {},
};
