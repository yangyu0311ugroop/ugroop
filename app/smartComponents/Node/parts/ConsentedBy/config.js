import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    value: NODE_STORE_SELECTORS.lastModifiedBy,
  },
};
