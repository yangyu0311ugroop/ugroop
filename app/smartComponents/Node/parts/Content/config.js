import { NODE_API, UPDATE_NODE } from 'apis/constants';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { ORGANISATION_STORE_SELECTORS } from 'datastore/orgStore/selectors';
import { RESAGA_HELPERS } from 'utils/helpers/resaga';

export const CONFIG = {
  value: {
    content: RESAGA_HELPERS.subscribeIfNotGiven(
      NODE_STORE_SELECTORS.content,
      'content',
    ),
    type: NODE_STORE_SELECTORS.type,
    status: NODE_STORE_SELECTORS.status,
    memberIds: ({ orgId }) =>
      ORGANISATION_STORE_SELECTORS.getRoleMembersIds({ id: orgId }),
  },
  setValue: {
    nodes: NODE_STORE_SELECTORS.nodes,
  },
  isLoading: {
    updatingNode: [NODE_API, UPDATE_NODE],
  },
};
