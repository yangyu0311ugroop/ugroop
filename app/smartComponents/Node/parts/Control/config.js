import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    control: NODE_STORE_SELECTORS.control,
  },
  setValue: {},
};
