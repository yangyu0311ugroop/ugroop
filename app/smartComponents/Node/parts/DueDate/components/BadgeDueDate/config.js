import { NODE_STORE_HELPERS } from 'datastore/nodeStore/helpers';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { DUE_DATE_HELPERS } from 'smartComponents/Node/parts/DueDate/components/ChangeDueDate/helpers';

export const CONFIG = {
  value: {
    parentType: NODE_STORE_SELECTORS.parentType,
    translateType: {
      keyPath: NODE_STORE_SELECTORS.parentType,
      getter: NODE_STORE_HELPERS.translateType,
    },
    dueDate: NODE_STORE_SELECTORS.dueDate,
    status: NODE_STORE_SELECTORS.status,
    calculatedDate: {
      getter: DUE_DATE_HELPERS.calculateDate,
    },
  },
  setValue: {
    calculatedStartTimeValue: NODE_STORE_SELECTORS.calculatedStartTimeValue,
  },
};
