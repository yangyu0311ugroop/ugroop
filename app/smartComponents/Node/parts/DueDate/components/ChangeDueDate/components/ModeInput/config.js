import { NODE_STORE_HELPERS } from 'datastore/nodeStore/helpers';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    type: NODE_STORE_SELECTORS.type,
    dueDate: NODE_STORE_SELECTORS.dueDate,
    parentType: NODE_STORE_SELECTORS.parentType,
    translateType: {
      keyPath: NODE_STORE_SELECTORS.parentType,
      getter: NODE_STORE_HELPERS.translateType,
    },
  },
  setValue: {},
};
