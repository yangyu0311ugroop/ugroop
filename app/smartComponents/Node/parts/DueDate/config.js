import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { RESAGA_HELPERS } from 'utils/helpers/resaga';
import { DASHBOARD_VIEW_STORE, NODE_STORE } from 'appConstants';

export const CONFIG = {
  value: {
    type: NODE_STORE_SELECTORS.type,
    dueDate: RESAGA_HELPERS.subscribeIfNotGiven(
      NODE_STORE_SELECTORS.dueDate,
      'dueDate',
    ),
    startTime: {
      cacheKey: ({ id }) => `${id}.startTime`,
      keyPath: NODE_STORE_SELECTORS.calculatedStartTime,
      props: null,
      getter: start => {
        if (typeof start !== 'object') return null;

        if (!start.real) return null;

        return start.value;
      },
    },
  },
};

export const CONFIG2 = {
  value: {
    timeNodesCount: {
      keyPath: ({ startTime }) => [NODE_STORE, 'timeNodes', [startTime]],
      getter: (nodes = []) => nodes.length,
    },
  },
  setValue: {
    timeNodes: [NODE_STORE, 'timeNodes'],
    times: [DASHBOARD_VIEW_STORE, 'times'],
  },
};
