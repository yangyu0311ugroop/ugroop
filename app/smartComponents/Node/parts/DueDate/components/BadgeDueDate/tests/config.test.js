import { CONFIG } from '../config';

describe('BadgeDueDate/config.js', () => {
  afterEach(() => jest.clearAllMocks());

  describe('Smoke Test', () => {
    it('should exists', () => {
      expect(CONFIG).toMatchSnapshot();
    });
  });
});
