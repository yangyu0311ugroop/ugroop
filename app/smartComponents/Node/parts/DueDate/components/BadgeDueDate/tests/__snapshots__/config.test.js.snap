// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeDueDate/config.js Smoke Test should exists 1`] = `
Object {
  "setValue": Object {
    "calculatedStartTimeValue": [Function],
  },
  "value": Object {
    "calculatedDate": Object {
      "getter": [Function],
    },
    "dueDate": [Function],
    "parentType": [Function],
    "status": [Function],
    "translateType": Object {
      "getter": [Function],
      "keyPath": [Function],
    },
  },
}
`;
